{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-09-21T23:02:01.622875Z",
     "start_time": "2019-09-21T23:02:01.451087Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      " USAGE:\r\n",
      "\r\n",
      "trim_galore [options] <filename(s)>\r\n",
      "\r\n",
      "\r\n",
      "-h/--help               Print this help message and exits.\r\n",
      "\r\n",
      "-v/--version            Print the version information and exits.\r\n",
      "\r\n",
      "-q/--quality <INT>      Trim low-quality ends from reads in addition to adapter removal. For\r\n",
      "                        RRBS samples, quality trimming will be performed first, and adapter\r\n",
      "                        trimming is carried in a second round. Other files are quality and adapter\r\n",
      "                        trimmed in a single pass. The algorithm is the same as the one used by BWA\r\n",
      "                        (Subtract INT from all qualities; compute partial sums from all indices\r\n",
      "                        to the end of the sequence; cut sequence at the index at which the sum is\r\n",
      "                        minimal). Default Phred score: 20.\r\n",
      "\r\n",
      "--phred33               Instructs Cutadapt to use ASCII+33 quality scores as Phred scores\r\n",
      "                        (Sanger/Illumina 1.9+ encoding) for quality trimming. Default: ON.\r\n",
      "\r\n",
      "--phred64               Instructs Cutadapt to use ASCII+64 quality scores as Phred scores\r\n",
      "                        (Illumina 1.5 encoding) for quality trimming.\r\n",
      "\r\n",
      "--fastqc                Run FastQC in the default mode on the FastQ file once trimming is complete.\r\n",
      "\r\n",
      "--fastqc_args \"<ARGS>\"  Passes extra arguments to FastQC. If more than one argument is to be passed\r\n",
      "                        to FastQC they must be in the form \"arg1 arg2 etc.\". An example would be:\r\n",
      "                        --fastqc_args \"--nogroup --outdir /home/\". Passing extra arguments will\r\n",
      "                        automatically invoke FastQC, so --fastqc does not have to be specified\r\n",
      "                        separately.\r\n",
      "\r\n",
      "-a/--adapter <STRING>   Adapter sequence to be trimmed. If not specified explicitly, Trim Galore will\r\n",
      "                        try to auto-detect whether the Illumina universal, Nextera transposase or Illumina\r\n",
      "                        small RNA adapter sequence was used. Also see '--illumina', '--nextera' and\r\n",
      "                        '--small_rna'. If no adapter can be detected within the first 1 million sequences\r\n",
      "                        of the first file specified Trim Galore defaults to '--illumina'.\r\n",
      "\r\n",
      "-a2/--adapter2 <STRING> Optional adapter sequence to be trimmed off read 2 of paired-end files. This\r\n",
      "                        option requires '--paired' to be specified as well. If the libraries to be trimmed\r\n",
      "                        are smallRNA then a2 will be set to the Illumina small RNA 5' adapter automatically\r\n",
      "                        (GATCGTCGGACT).\r\n",
      "\r\n",
      "--illumina              Adapter sequence to be trimmed is the first 13bp of the Illumina universal adapter\r\n",
      "                        'AGATCGGAAGAGC' instead of the default auto-detection of adapter sequence.\r\n",
      "\r\n",
      "--nextera               Adapter sequence to be trimmed is the first 12bp of the Nextera adapter\r\n",
      "                        'CTGTCTCTTATA' instead of the default auto-detection of adapter sequence.\r\n",
      "\r\n",
      "--small_rna             Adapter sequence to be trimmed is the first 12bp of the Illumina Small RNA 3' Adapter\r\n",
      "                        'TGGAATTCTCGG' instead of the default auto-detection of adapter sequence. Selecting\r\n",
      "                        to trim smallRNA adapters will also lower the --length value to 18bp. If the smallRNA\r\n",
      "                        libraries are paired-end then a2 will be set to the Illumina small RNA 5' adapter\r\n",
      "                        automatically (GATCGTCGGACT) unless -a 2 had been defined explicitly.\r\n",
      "\r\n",
      "--max_length <INT>      Discard reads that are longer than <INT> bp after trimming. This is only advised for\r\n",
      "                        smallRNA sequencing to remove non-small RNA sequences.\r\n",
      "\r\n",
      "\r\n",
      "--stringency <INT>      Overlap with adapter sequence required to trim a sequence. Defaults to a\r\n",
      "                        very stringent setting of 1, i.e. even a single bp of overlapping sequence\r\n",
      "                        will be trimmed off from the 3' end of any read.\r\n",
      "\r\n",
      "-e <ERROR RATE>         Maximum allowed error rate (no. of errors divided by the length of the matching\r\n",
      "                        region) (default: 0.1)\r\n",
      "\r\n",
      "--gzip                  Compress the output file with GZIP. If the input files are GZIP-compressed\r\n",
      "                        the output files will automatically be GZIP compressed as well. As of v0.2.8 the\r\n",
      "                        compression will take place on the fly.\r\n",
      "\r\n",
      "--dont_gzip             Output files won't be compressed with GZIP. This option overrides --gzip.\r\n",
      "\r\n",
      "--length <INT>          Discard reads that became shorter than length INT because of either\r\n",
      "                        quality or adapter trimming. A value of '0' effectively disables\r\n",
      "                        this behaviour. Default: 20 bp.\r\n",
      "\r\n",
      "                        For paired-end files, both reads of a read-pair need to be longer than\r\n",
      "                        <INT> bp to be printed out to validated paired-end files (see option --paired).\r\n",
      "                        If only one read became too short there is the possibility of keeping such\r\n",
      "                        unpaired single-end reads (see --retain_unpaired). Default pair-cutoff: 20 bp.\r\n",
      "\r\n",
      "--max_n COUNT           The total number of Ns (as integer) a read may contain before it will be removed altogether.\r\n",
      "                        In a paired-end setting, either read exceeding this limit will result in the entire\r\n",
      "                        pair being removed from the trimmed output files.\r\n",
      "\r\n",
      "--trim-n                Removes Ns from either side of the read. This option does currently not work in RRBS mode.\r\n",
      "\r\n",
      "-o/--output_dir <DIR>   If specified all output will be written to this directory instead of the current\r\n",
      "                        directory.\r\n",
      "\r\n",
      "--no_report_file        If specified no report file will be generated.\r\n",
      "\r\n",
      "--suppress_warn         If specified any output to STDOUT or STDERR will be suppressed.\r\n",
      "\r\n",
      "--clip_R1 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 1 (or single-end\r\n",
      "                        reads). This may be useful if the qualities were very poor, or if there is some\r\n",
      "                        sort of unwanted bias at the 5' end. Default: OFF.\r\n",
      "\r\n",
      "--clip_R2 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 2 (paired-end reads\r\n",
      "                        only). This may be useful if the qualities were very poor, or if there is some sort\r\n",
      "                        of unwanted bias at the 5' end. For paired-end BS-Seq, it is recommended to remove\r\n",
      "                        the first few bp because the end-repair reaction may introduce a bias towards low\r\n",
      "                        methylation. Please refer to the M-bias plot section in the Bismark User Guide for\r\n",
      "                        some examples. Default: OFF.\r\n",
      "\r\n",
      "--three_prime_clip_R1 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 1 (or single-end\r\n",
      "                        reads) AFTER adapter/quality trimming has been performed. This may remove some unwanted\r\n",
      "                        bias from the 3' end that is not directly related to adapter sequence or basecall quality.\r\n",
      "                        Default: OFF.\r\n",
      "\r\n",
      "--three_prime_clip_R2 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 2 AFTER\r\n",
      "                        adapter/quality trimming has been performed. This may remove some unwanted bias from\r\n",
      "                        the 3' end that is not directly related to adapter sequence or basecall quality.\r\n",
      "                        Default: OFF.\r\n",
      "\r\n",
      "--path_to_cutadapt </path/to/cutadapt>     You may use this option to specify a path to the Cutadapt executable,\r\n",
      "                        e.g. /my/home/cutadapt-1.7.1/bin/cutadapt. Else it is assumed that Cutadapt is in\r\n",
      "                        the PATH.\r\n",
      "\r\n",
      "\r\n",
      "RRBS-specific options (MspI digested material):\r\n",
      "\r\n",
      "--rrbs                  Specifies that the input file was an MspI digested RRBS sample (recognition\r\n",
      "                        site: CCGG). Single-end or Read 1 sequences (paired-end) which were adapter-trimmed\r\n",
      "                        will have a further 2 bp removed from their 3' end. Sequences which were merely\r\n",
      "                        trimmed because of poor quality will not be shortened further. Read 2 of paired-end\r\n",
      "                        libraries will in addition have the first 2 bp removed from the 5' end (by setting \r\n",
      "                        '--clip_r2 2'). This is to avoid using artificial methylation calls from the filled-in\r\n",
      "                        cytosine positions close to the 3' MspI site in sequenced fragments. \r\n",
      "                        This option is not recommended for users of the NuGEN ovation RRBS System 1-16\r\n",
      "                        kit (see below).\r\n",
      "\r\n",
      "--non_directional       Selecting this option for non-directional RRBS libraries will screen\r\n",
      "                        quality-trimmed sequences for 'CAA' or 'CGA' at the start of the read\r\n",
      "                        and, if found, removes the first two basepairs. Like with the option\r\n",
      "                        '--rrbs' this avoids using cytosine positions that were filled-in\r\n",
      "                        during the end-repair step. '--non_directional' requires '--rrbs' to\r\n",
      "                        be specified as well. Note that this option does not set '--clip_r2 2' in\r\n",
      "                        paired-end mode.\r\n",
      "\r\n",
      "--keep                  Keep the quality trimmed intermediate file. Default: off, which means\r\n",
      "                        the temporary file is being deleted after adapter trimming. Only has\r\n",
      "                        an effect for RRBS samples since other FastQ files are not trimmed\r\n",
      "                        for poor qualities separately.\r\n",
      "\r\n",
      "\r\n",
      "Note for RRBS using the NuGEN Ovation RRBS System 1-16 kit:\r\n",
      "\r\n",
      "Owing to the fact that the NuGEN Ovation kit attaches a varying number of nucleotides (0-3) after each MspI\r\n",
      "site Trim Galore should be run WITHOUT the option --rrbs. This trimming is accomplished in a subsequent \r\n",
      "diversity trimming step afterwards (see their manual).\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Note for RRBS using MseI:\r\n",
      "\r\n",
      "If your DNA material was digested with MseI (recognition motif: TTAA) instead of MspI it is NOT necessary\r\n",
      "to specify --rrbs or --non_directional since virtually all reads should start with the sequence\r\n",
      "'TAA', and this holds true for both directional and non-directional libraries. As the end-repair of 'TAA'\r\n",
      "restricted sites does not involve any cytosines it does not need to be treated especially. Instead, simply\r\n",
      "run Trim Galore! in the standard (i.e. non-RRBS) mode.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Paired-end specific options:\r\n",
      "\r\n",
      "--paired                This option performs length trimming of quality/adapter/RRBS trimmed reads for\r\n",
      "                        paired-end files. To pass the validation test, both sequences of a sequence pair\r\n",
      "                        are required to have a certain minimum length which is governed by the option\r\n",
      "                        --length (see above). If only one read passes this length threshold the\r\n",
      "                        other read can be rescued (see option --retain_unpaired). Using this option lets\r\n",
      "                        you discard too short read pairs without disturbing the sequence-by-sequence order\r\n",
      "                        of FastQ files which is required by many aligners.\r\n",
      "\r\n",
      "                        Trim Galore! expects paired-end files to be supplied in a pairwise fashion, e.g.\r\n",
      "                        file1_1.fq file1_2.fq SRR2_1.fq.gz SRR2_2.fq.gz ... .\r\n",
      "\r\n",
      "-t/--trim1              Trims 1 bp off every read from its 3' end. This may be needed for FastQ files that\r\n",
      "                        are to be aligned as paired-end data with Bowtie. This is because Bowtie (1) regards\r\n",
      "                        alignments like this:\r\n",
      "\r\n",
      "                          R1 --------------------------->     or this:    ----------------------->  R1\r\n",
      "                          R2 <---------------------------                       <-----------------  R2\r\n",
      "\r\n",
      "                        as invalid (whenever a start/end coordinate is contained within the other read).\r\n",
      "                        NOTE: If you are planning to use Bowtie2, BWA etc. you don't need to specify this option.\r\n",
      "\r\n",
      "--retain_unpaired       If only one of the two paired-end reads became too short, the longer\r\n",
      "                        read will be written to either '.unpaired_1.fq' or '.unpaired_2.fq'\r\n",
      "                        output files. The length cutoff for unpaired single-end reads is\r\n",
      "                        governed by the parameters -r1/--length_1 and -r2/--length_2. Default: OFF.\r\n",
      "\r\n",
      "-r1/--length_1 <INT>    Unpaired single-end read length cutoff needed for read 1 to be written to\r\n",
      "                        '.unpaired_1.fq' output file. These reads may be mapped in single-end mode.\r\n",
      "                        Default: 35 bp.\r\n",
      "\r\n",
      "-r2/--length_2 <INT>    Unpaired single-end read length cutoff needed for read 2 to be written to\r\n",
      "                        '.unpaired_2.fq' output file. These reads may be mapped in single-end mode.\r\n",
      "                        Default: 35 bp.\r\n",
      "\r\n",
      "\r\n",
      "Last modified on 13 November 2017.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/TrimGalore-0.4.5/trim_galore -help\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-09-21T23:19:08.597Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.16\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count smallRNA: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz: uncompress failed\n",
      "Writing report to '/Volumes/block/caligus/trim/Sealice_F1_S20_L001_R1_001_val_1.fq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.4_dev\n",
      "Cutadapt version: 1.16\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "All Read 1 sequences will be trimmed by 7 bp from their 5' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 7 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n",
      "All Read 1 sequences will be trimmed by 7 bp from their 3' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 7 bp from their 3' end to avoid poor qualities or biases\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to Sealice_F1_S20_L001_R1_001_val_1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz <<< \n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/TrimGalore-0.4.5/trim_galore \\\n",
    "--paired \\\n",
    "--clip_R1 7 \\\n",
    "--clip_R2 7 \\\n",
    "--three_prime_clip_R1 7 \\\n",
    "--three_prime_clip_R2 7 \\\n",
    "--output_dir /Volumes/block/caligus/trim/ \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R2_001_val_2.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L002_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L002_R2_001_val_2.fq.gz \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-09-21T23:20:49.366Z"
    }
   },
   "outputs": [],
   "source": [
    "!/Applications/bioinfo/TrimGalore-0.4.5/trim_galore \\\n",
    "--paired \\\n",
    "--clip_R1 7 \\\n",
    "--clip_R2 7 \\\n",
    "--three_prime_clip_R1 7 \\\n",
    "--three_prime_clip_R2 7 \\\n",
    "--output_dir /Volumes/block/caligus/trim/ \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L001_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L001_R2_001_val_2.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L002_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L002_R2_001_val_2.fq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
