{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "## Taking the coverage file from an alignment of all BS data to genome, will destrand."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-06T00:52:15.733763Z",
     "start_time": "2019-11-06T00:52:15.656154Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/sr320/Documents/GitHub/nb-2019/P_generosa'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/Serine/wd/19-11-05/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "This script is supposed to convert a specified reference genome into two different bisulfite\r\n",
      "converted versions and index them for alignments with Bowtie 2 (default), or Bowtie 1. The first\r\n",
      "bisulfite genome will have all Cs converted to Ts (C->T), and the other one will have all Gs\r\n",
      "converted to As (G->A). Both bisulfite genomes will be stored in subfolders within the reference\r\n",
      "genome folder. Once the bisulfite conversion has been completed the program will fork and launch\r\n",
      "two simultaneous instances of the Bowtie 1 or 2 indexer (bowtie-build or bowtie2-build). Be aware\r\n",
      "that the indexing process can take up to several hours; this will mainly depend on genome size\r\n",
      "and system resources.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "The following is a brief description of command line options and arguments to control the\r\n",
      "Bismark Genome Preparation:\r\n",
      "\r\n",
      "\r\n",
      "USAGE: bismark_genome_preparation [options] <arguments>\r\n",
      "\r\n",
      "\r\n",
      "OPTIONS:\r\n",
      "\r\n",
      "--help/--man             Displays this help filea and exits.\r\n",
      "\r\n",
      "--version                Displays version information and exits.\r\n",
      "\r\n",
      "--verbose                Print verbose output for more details or debugging.\r\n",
      "\r\n",
      "--path_to_bowtie </../>  The full path to the Bowtie 1 or Bowtie 2 installation on your system\r\n",
      "                         (depending on which aligner/indexer you intend to use). Unless this path\r\n",
      "                         is specified it is assumed that Bowtie is in the PATH.\r\n",
      "\r\n",
      "--bowtie2                This will create bisulfite indexes for Bowtie 2. (Default: ON).\r\n",
      "\r\n",
      "--bowtie1                This will create bisulfite indexes for Bowtie 1. (Default: OFF).\r\n",
      "\r\n",
      "--single_fasta           Instruct the Bismark Indexer to write the converted genomes into\r\n",
      "                         single-entry FastA files instead of making one multi-FastA file (MFA)\r\n",
      "                         per chromosome. This might be useful if individual bisulfite converted\r\n",
      "                         chromosomes are needed (e.g. for debugging), however it can cause a\r\n",
      "                         problem with indexing if the number of chromosomes is vast (this is likely\r\n",
      "                         to be in the range of several thousand files; the operating system can\r\n",
      "                         only handle lists up to a certain length, and some newly assembled\r\n",
      "                         genomes may contain 20000-500000 contigs of scaffold files which do exceed\r\n",
      "                         this list length limit).\r\n",
      "\r\n",
      "--genomic_composition    Calculate and extract the genomic sequence composition for mono and di-nucleotides\r\n",
      "                         and write the genomic composition table 'genomic_nucleotide_frequencies.txt' to the\r\n",
      "                         genome folder. This may be useful later on when using bam2nuc or the Bismark option\r\n",
      "                         --nucleotide_coverage.\r\n",
      "\r\n",
      "\r\n",
      "ARGUMENTS:\r\n",
      "\r\n",
      "<path_to_genome_folder>  The path to the folder containing the genome to be bisulfite converted.\r\n",
      "                         The Bismark Genome Preparation expects one or more fastA files in the folder\r\n",
      "                         (with the file extension: .fa or .fasta (also ending in .gz)). Specifying this path is mandatory.\r\n",
      "\r\n",
      "\r\n",
      "This script was last modified on 26 April 2018.\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/bismark_genome_preparation -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path to genome folder specified as: /Volumes/Serine/wd/19-11-05/\n",
      "Aligner to be used: Bowtie 2 (default)\n",
      "Writing bisulfite genomes out into a single MFA (multi FastA) file\n",
      "\n",
      "Bismark Genome Preparation - Step I: Preparing folders\n",
      "\n",
      "Path to Bowtie 2 specified: /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/\n",
      "Bisulfite Genome Indexer version v0.20.0 (last modified 26 April 2018)\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/CT_conversion/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/GA_conversion/\n",
      "\n",
      "Step I - Prepare genome folders - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step II: Bisulfite converting reference genome\n",
      "\n",
      "conversions performed:\n",
      "chromosome\tC->T\tG->A\n",
      "Scaffold_01\t12767429\t12765526\n",
      "Scaffold_02\t10088811\t10062918\n",
      "Scaffold_03\t8199278\t8194496\n",
      "Scaffold_04\t9161562\t9161987\n",
      "Scaffold_05\t9498414\t9497714\n",
      "Scaffold_06\t8458182\t8463483\n",
      "Scaffold_07\t6054340\t6047203\n",
      "Scaffold_08\t8517146\t8506733\n",
      "Scaffold_09\t5420328\t5445998\n",
      "Scaffold_10\t7547438\t7543051\n",
      "Scaffold_11\t7053984\t7054209\n",
      "Scaffold_12\t7145712\t7105651\n",
      "Scaffold_13\t6273846\t6290785\n",
      "Scaffold_14\t6515251\t6527291\n",
      "Scaffold_15\t6625508\t6624214\n",
      "Scaffold_16\t4528254\t4550625\n",
      "Scaffold_17\t4780809\t4787183\n",
      "Scaffold_18\t3756408\t3758031\n",
      "\n",
      "Total number of conversions performed:\n",
      "C->T:\t132392700\n",
      "G->A:\t132387098\n",
      "\n",
      "Step II - Genome bisulfite conversions - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step III: Launching the Bowtie 2 indexer\n",
      "Please be aware that this process can - depending on genome size - take several hours!\n",
      "\n",
      "Preparing indexing of CT converted genome in /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/CT_conversion/\n",
      "Parent process: Starting to index C->T converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.CT_conversion.fa BS_CT\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_CT.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.CT_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "Preparing indexing of GA converted genome in /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/GA_conversion/\n",
      "Child process: Starting to index G->A converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.GA_conversion.fa BS_GA\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_GA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.GA_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:31\n",
      "Returning block of 783861162 for bucket 1\n",
      "  Sorting block time: 00:09:24\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_GA.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:10:18\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:07\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:17\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_CT.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:12:15\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:07\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/bismark_genome_preparation \\\n",
    "--verbose \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/ \\\n",
    "/Volumes/Serine/wd/19-11-05/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine \\\n",
    "--genome_folder /Volumes/Serine/wd/19-11-05/ \\\n",
    "-o /Volumes/Serine/wd/19-11-05/ \\\n",
    "--merge_CpG \\\n",
    "/Volumes/Serine/wd/19-11-05/Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
