---
title: "52-methykit"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Obtain session information

```{r}
sessionInfo()
```


Ambient- 8	Day10 

compared to

Super.Low	-7 Day10

![img](https://gannet.fish.washington.edu/seashell/snaps/genome_memory_manuscript_-_Google_Docs_2019-09-13_10-25-37.png)


```{r}
library(tidyverse)

```

```{r}
episamples <- read_csv("data/sample.info.csv")
```


```{r}
filter(episamples, TimePoint == "Day10", Initial.Treatment != "Low")
```


```{bash}
ls /Volumes/block/geoduck/082119_bam/

```





```{r}
file.list_d10=list('/Volumes/block/geoduck/082119_bam/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam',
                 '/Volumes/block/geoduck/082119_bam/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam',
                 '/Volumes/block/geoduck/082119_bam/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam',
                 '/Volumes/block/geoduck/082119_bam/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam',
                 '/Volumes/block/geoduck/082119_bam/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam',
                 '/Volumes/block/geoduck/082119_bam/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam',
                 '/Volumes/block/geoduck/082119_bam/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam',
                 '/Volumes/block/geoduck/082119_bam/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.sorted.bam'
)
```





```{r}
library(methylKit)
```



```{r, eval = FALSE}
myobj_d10 = processBismarkAln(location = file.list_d10, sample.id = list("111","113","143","145","119","120","135","136"), assembly = "v074", read.context="CpG", mincov=2, treatment = c(0,0,0,0,1,1,1,1))
```

```{r, eval = FALSE}
save(myobj_d10, file = "analyses/myobj_d10")
```



```{r}
View(getData(myobj_d10[[1]]))
```

```{r}
View(getData(myobj_d10[[2]]))
```



available at http://gannet.fish.washington.edu/seashell/snaps/myobj_d10







```{bash}
zip ../analyses/myobj_d10.zip ../analyses/myobj_d10
```



```{r}
load("analyses/myobj_d10")
```





```{r}
getMethylationStats(myobj_d10[[2]],plot=FALSE,both.strands=FALSE)
```

```{r}
getMethylationStats(myobj_d10
                    [[8]],plot=TRUE,both.strands=FALSE)
```

```{r}
getCoverageStats(myobj_d10[[5]],plot=TRUE,both.strands=FALSE)
```


```{r}
filtered.myobj=filterByCoverage(myobj_d10,lo.count=5,lo.perc=NULL,
                                      hi.count=NULL,hi.perc=99)
meth_filter=methylKit::unite(filtered.myobj, destrand=TRUE)
clusterSamples(meth_filter, dist="correlation", method="ward", plot=TRUE)
PCASamples(meth_filter)
```






```{r}
View(getData(meth_filter))
```











```{r}
myDiff=calculateDiffMeth(meth_filter,mc.cores=8)
```



```{r}
# get hyper methylated bases
myDiff25p.hyper=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hyper")
#
# get hypo methylated bases
myDiff25p.hypo=getMethylDiff(myDiff,difference=25,qvalue=0.01,type="hypo")
#
#
# get all differentially methylated bases
myDiff25p=getMethylDiff(myDiff,difference=25,qvalue=0.01)
```




```{r}
write.table(myDiff25p, file = "analyses/myDiff25p.tab", sep = "\t")
```


```{r}
# get hyper methylated bases
myDiff75p.hyper=getMethylDiff(myDiff,difference=75,qvalue=0.01,type="hyper")
#
# get hypo methylated bases
myDiff75p.hypo=getMethylDiff(myDiff,difference=75,qvalue=0.01,type="hypo")
#
#
# get all differentially methylated bases
myDiff75p=getMethylDiff(myDiff,difference=75,qvalue=0.01)
```




```{r}
write.table(myDiff75p, file = "analyses/myDiff75p_d10.tab",  quote = FALSE, sep = "\t", row.names=FALSE)

```


```{bash}
head analyses/myDiff75p_d10.tab

```






```{r}
# get hyper methylated bases
myDiff50p.hyper=getMethylDiff(myDiff,difference=50,qvalue=0.01,type="hyper")
#
# get hypo methylated bases
myDiff50p.hypo=getMethylDiff(myDiff,difference=50,qvalue=0.01,type="hypo")
#
#
# get all differentially methylated bases
myDiff50p=getMethylDiff(myDiff,difference=50,qvalue=0.01)
```


```{r}
# get hyper methylated bases
myDiff100p.hyper=getMethylDiff(myDiff,difference=50,qvalue=0.01,type="hyper")
#
# get hypo methylated bases
myDiff100p.hypo=getMethylDiff(myDiff,difference=50,qvalue=0.01,type="hypo")
#
#
# get all differentially methylated bases
myDiff100p=getMethylDiff(myDiff,difference=50,qvalue=0.01)
```



```{r}
write.table(myDiff50p, file = "analyses/myDiff50p_d10.tab", sep = "\t")
```




---
# Taking the DMLs to a bed

```{r}
library(readr)
myDiff50p_d10 <- read_tsv("analyses/myDiff50p_d10.tab")
 
```


```{r}
view(myDiff50p_d10)
```

```{r}
head(myDiff50p_d10)
```



```{r}
dml_d10 <-  mutate(myDiff50p_d10, start = start -1, end = end + 1) %>% select(chr, start, end, meth.diff) %>% 
  mutate_if(is.numeric, as.integer) 
write_delim(dml_d10, "analyses/dml_d10.bed",  delim = '\t', col_names = FALSE)
```


```{r}
head(dml_d10)
```


```{bash}
/Applications/bioinfo/bedtools2/bin/intersectBed \
-u \
-a analyses/dml_d10.bed \
-b tracks/Panopea-generosa-vv0.74.a3.gene.gff3 \
| wc -l
```




#75p bed

```{r}
library(readr)
myDiffp75_d10 <- read_tsv("analyses/myDiff75p_d10.tab")
 
```

```{r}
library(dplyr)
```




```{r}
dml75_d10 <-  mutate(myDiffp75_d10, start = start -1, end = end + 1) %>% select(chr, start, end, meth.diff) %>% 
  mutate_if(is.numeric, as.integer) 
write_delim(dml75_d10, "analyses/dml75_d10.bed",  delim = '\t', col_names = FALSE)
```




```{bash}
/Applications/bioinfo/bedtools2/bin/intersectBed \
-u \
-a analyses/dml75_d10.bed \
-b tracks/Panopea-generosa-vv0.74.a3.gene.gff3 \
| wc -l
```
