---
title: "56-karotypR-tutorial"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
 gains <- toGRanges(data.frame(chr=c("chr1", "chr5", "chr17", "chr22"), start=c(1, 1000000, 4000000, 1),
                      end=c(5000000, 3200000, 80000000, 1200000)))

  losses <- toGRanges(data.frame(chr=c("chr3", "chr9", "chr17"), start=c(80000000, 20000000, 1),
                       end=c(170000000, 30000000, 25000000)))
  
  kp <- plotKaryotype(genome="hg19")
  
  kpPlotRegions(kp, gains, col="#FFAACC")
  
  kpPlotRegions(kp, losses, col="#CCFFAA")
```


```{r}
 g.gains <- toGRanges(data.frame(chr=c("1", "1", "5", "17"), start=c(1, 7000000, 1000, 4000000),
                      end=c(5000000, 90000000, 3200000, 80000000)))
  
  kp <- plotKaryotype(genome= duck.genome)
  kpPlotRegions(kp, g.gains, col="blue")
  
head(g.gains)  
```


```{bash}
head tracks/Panopea-generosa-vv0.74.a3.intron.bed
```


```{bash}
sed 's/PGA_scaffold1__77_contigs__length_89643857/1/g' tracks/Panopea-generosa-vv0.74.a3.intron.bed > analyses/_test.1.bed

head tracks/Panopea-generosa-vv0.74.a3.intron.bed
head analyses/_test.1.bed

```

```{bash}
tail -1 tracks/Panopea-generosa-vv0.74.a3.gene.gff3

awk -F"\t" 'NR>3 {print $1"\t",$4"\t", $5}' tracks/Panopea-generosa-vv0.74.a3.gene.gff3 |  head


awk -F"\t" 'NR>3 {print $1"\t",$4"\t", $5}' tracks/Panopea-generosa-vv0.74.a3.gene.gff3 | sed 's/PGA_scaffold1__77_contigs__length_89643857/1/g' | head

awk -F"\t" 'NR>3 {print $1"\t",$4"\t", $5}' tracks/Panopea-generosa-vv0.74.a3.gene.gff3 | sed 's/PGA_scaffold1__77_contigs__length_89643857/1/g' > analyses/_test.gene.1.bed

```



```{r}
library(rtracklayer)
gr_obj =  import("analyses/_test.1.bed")

gene_obj = import("analyses/_test.gene.1.bed")
  
  kp <- plotKaryotype(genome= duck.genome)
  kpPlotRegions(kp, gene_obj, col="blue")
```


  





```{r}
  kp <- plotKaryotype(genome="mm10", plot.type=1, main="The mm10 genome")

```


```{r}




kp <- plotKaryotype(genome= duck.genome)




kpPlotDensity(kp, data=gene_obj,window.size = 100000, col=adjustcolor("blue", 0.5), lwd=0.5, border=adjustcolor("blue", 0.5), r0=0, r1=1)  

  
```


```{bash}


cat tracks/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph | sed 's/PGA_scaffold1__77_contigs__length_89643857/1/g' > analyses/_test.bedGraph

head analyses/_test.bedGraph
```

```{bash}
head analyses/_testv2.bedGraph
```



```{r}
test_obj = import.bedGraph("analyses/_testv2.bedGraph")
```

```{r}
dmr_obj = import("analyses/dmr.bed")
```



```{r}
head(test_obj)
```

#filtering

```{r}
library("tidyverse")

```


```{r}
testv2.bedGraph <- read.csv("analyses/_testv2.bedGraph", sep = "\t", header = FALSE)

view(testv2.bedGraph)


testv3.bedGraph <-filter(testv2.bedGraph, V4 > 75)
```

```{r}
write_delim(testv3.bedGraph, "analyses/testv3.bedGraph", delim = "\t", col_names = FALSE)
```


```{r}

test3_obj = import.bedGraph("analyses/testv3.bedGraph")
```

```{r}
head(test3_obj)
```



kp <- plotKaryotype(genome=CV.genome[1:10], ideogram.plotter = NULL, chromosomes=chrs, plot.type=2)
kpAddCytobandsAsLine(kp)
kpPlotDensity(kp, data=exons,window.size = 100000, col=adjustcolor("blue", 0.5), lwd=0.5, border=adjustcolor("blue", 0.5), data.panel=1)
kpPlotDensity(kp, data=genes,window.size = 100000, col=adjustcolor("red", 0.3), lwd=0.5, data.panel=2) #this should plot on the bottom, but doesn't seem to work

```{r}
pp <- getDefaultPlotParams(plot.type=2)
pp

```

```{r}
pp$ideogramheight <- 50
pp$dataallmax <- 1
```



```{r}
kp <- plotKaryotype(genome=duck.genome, plot.type=2, chromosomes=c("1", "2", "18"))
  
  kpPlotDensity(kp, data=gene_obj,window.size = 10000, data.panel=2, col=adjustcolor("blue", 0.5), lwd=0.5, border=adjustcolor("blue", 0.5), r0=.5, r1=1)  

  #kpDataBackground(kp, data.panel = 1, r0=0, r1=0.45)
  kpAxis(kp, ymin=0, ymax=100, r0=NULL, r1=NULL, data.panel=1, col="gray50", cex=0.5)
  
  #kpPoints(kp, chr=rand.datxa$chr, x=rand.data$start, y=rand.data$y,
           #ymin=-1, ymax=1, r0=0.05, r1=0.4, col="black", pch=".", cex=2)
  
  kpPoints(kp, data=test3_obj, x=test3_obj$ranges, y=test3_obj$score,
           ymin=0, ymax=100, data.panel=1, r0=NULL, r1=NULL, col=adjustcolor("pink", 0.5), pch=".", cex=0.2)
  
  kpPlotDensity(kp, data=test3_obj,window.size = 1000, data.panel=2, col=adjustcolor("green", 0.5), lwd=0.5, border=adjustcolor("green", 0.5), r0=0, r1=1) 
  
  
  kpPlotRegions(kp, dmr_obj, col="blue", r0=0, r1=.2, avoid.overlapping = TRUE, num.layers = 2)
  
   kpPlotCoverage(kp, regs, ymax = 20, r0=0.8, data.panel=2, r1=1, col="#CCCCFF")

  

```



```{r}
  kp <- plotKaryotype(genome=duck.genome, plot.type=1, main="The mm10 genome")

```



```{r}
 kp <- plotKaryotype(genome="hg19", plot.type=2, chromosomes=c("chr1", "chr2", "chr3"))
```

```{r}

 kp <- plotKaryotype(genome=duck.genome, plot.type=2, chromosomes=c("1", "2", "17"))

```

```{r}
 rand.data <- createRandomRegions(genome=duck.genome, nregions=1000, length.mean=1, length.sd=0,
                      mask=NA, non.overlapping=TRUE) 
```

```{r}
rand.data <- toDataframe(sort(rand.data))
  rand.data <- cbind(rand.data, y=runif(n=1000, min=-1, max=1))
  
  #Select some data points as "special ones"
  sel.data <- rand.data[c(7, 30, 38, 52),] 
  head(rand.data)
```

```{r}



  kp <- plotKaryotype(genome=duck.genome, plot.type=2, chromosomes=c("1", "2", "18"))
  
  kpDataBackground(kp, data.panel = 1, r0=0, r1=0.45)
  kpAxis(kp, ymin=-1, ymax=1, r0=0.05, r1=0.4, col="gray50", cex=0.5)
  kpPoints(kp, chr=rand.data$chr, x=rand.data$start, y=rand.data$y,
           ymin=-1, ymax=1, r0=0.05, r1=0.4, col="black", pch=".", cex=2)
  kpPoints(kp, chr=sel.data$chr, x=sel.data$start, y=sel.data$y,
           ymin=-1, ymax=1, r0=0.05, r1=0.4, col="red")
  kpText(kp, chr=sel.data$chr, x=sel.data$start, y=sel.data$y,
         ymin=-1, ymax=1, r0=0.05, r1=0.4, labels=c("A", "B", "C", "D"), col="red",
         pos=4, cex=0.8)
  
  
  #Upper part: data.panel=1
  kpDataBackground(kp, data.panel = 1, r0=0.5, r1=1)
  kpAxis(kp, ymin=-1, ymax=1, r0=0.5, r1=1, col="gray50", cex=0.5, numticks = 5)
  kpAbline(kp, h=c(-0.5, 0, 0.5), col="gray50", ymin=-1, ymax=1, r0=0.5, r1=1)
  kpLines(kp, chr=rand.data$chr, x=rand.data$start, y=rand.data$y,
          col="#AA88FF", ymin=-1, ymax=1, r0=0.5, r1=1)
  #Use kpSegments to add small tic to the line
  kpSegments(kp, chr=rand.data$chr, x0=rand.data$start, x1=rand.data$start,
             y0=rand.data$y-0.1, y1=rand.data$y+0.1,
             col="#8866DD", ymin=-1, ymax=1, r0=0.5, r1=1)
  #Plot the same line but inverting the data by pssing a r0>r1
  kpLines(kp, chr=rand.data$chr, x=rand.data$start, y=rand.data$y,
          col="#FF88AA", ymin=-1, ymax=1, r0=1, r1=0.5)
  
  
  #Lower part: data.panel=2
  kpDataBackground(kp, r0=0, r1=0.29, color = "#EEFFEE", data.panel = 2)
  kpAxis(kp, col="#AADDAA", ymin=-1, ymax=1, r0=0, r1=0.29, data.panel = 2,
         numticks = 2, cex=0.5, tick.len = 0)
  kpAbline(kp, h=0, col="#AADDAA", ymin=-1, ymax=1, r0=0, r1=0.29, data.panel = 2)
  kpBars(kp, chr=rand.data$chr, x0=rand.data$start, x1=rand.data$end, y1 = rand.data$y,
          col="#AADDAA", ymin=-1, ymax=1, r0=0, r1=0.29, data.panel = 2, border="#AADDAA" )
  
  kpDataBackground(kp, r0=0.34, r1=0.63, color = "#EEEEFF", data.panel = 2)
  kpAxis(kp, col="#AAAADD", ymin=-1, ymax=1, r0=0.34, r1=0.63, data.panel = 2, 
         numticks = 2, cex=0.5, tick.len = 0)
  kpAbline(kp, h=0, col="#AAAADD", ymin=-1, ymax=1, r0=0.34, r1=0.63, data.panel = 2)
  kpSegments(kp, chr=rand.data$chr, x0=rand.data$start, x1=rand.data$end, 
             y0=rand.data$y-0.2, y1=rand.data$y, 
             col="#AAAADD", ymin=-1, ymax=1, r0=0.34, r1=0.63, data.panel = 2, lwd=2)
  
  kpDataBackground(kp, r0=0.68, r1=0.97, color = "#FFEEEE", data.panel = 2)
  kpAxis(kp, col="#DDAAAA", ymin=-1, ymax=1, r0=0.68, r1=0.97, data.panel = 2,
         numticks = 2, cex=0.5, tick.len = 0)
  kpPoints(kp, chr=rand.data$chr, x=rand.data$start, y=rand.data$y,
          col="#DDAAAA", ymin=-1, ymax=1, r0=0.68, r1=0.97, data.panel = 2, pch=".", cex=3)
  
```  
```{r}
 kp <- plotKaryotype(genome=duck.genome, plot.type=2, chromosomes=c("1", "2", "18"))
  
  kpDataBackground(kp, data.panel = 1, r0=0, r1=0.45)
  kpAxis(kp, ymin=-1, ymax=1, r0=0.05, r1=0.4, col="gray50", cex=0.5)
  kpPoints(kp, chr=rand.data$chr, x=rand.data$start, y=rand.data$y,
           ymin=-1, ymax=1, r0=0.05, r1=0.4, col="black", pch=".", cex=2)
  kpPoints(kp, chr=sel.data$chr, x=sel.data$start, y=sel.data$y,
           ymin=-1, ymax=1, r0=0.05, r1=0.4, col="red")
  
head(rand.data)
head(sel.data)
```

