{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-09-21T23:02:01.622875Z",
     "start_time": "2019-09-21T23:02:01.451087Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      " USAGE:\r\n",
      "\r\n",
      "trim_galore [options] <filename(s)>\r\n",
      "\r\n",
      "\r\n",
      "-h/--help               Print this help message and exits.\r\n",
      "\r\n",
      "-v/--version            Print the version information and exits.\r\n",
      "\r\n",
      "-q/--quality <INT>      Trim low-quality ends from reads in addition to adapter removal. For\r\n",
      "                        RRBS samples, quality trimming will be performed first, and adapter\r\n",
      "                        trimming is carried in a second round. Other files are quality and adapter\r\n",
      "                        trimmed in a single pass. The algorithm is the same as the one used by BWA\r\n",
      "                        (Subtract INT from all qualities; compute partial sums from all indices\r\n",
      "                        to the end of the sequence; cut sequence at the index at which the sum is\r\n",
      "                        minimal). Default Phred score: 20.\r\n",
      "\r\n",
      "--phred33               Instructs Cutadapt to use ASCII+33 quality scores as Phred scores\r\n",
      "                        (Sanger/Illumina 1.9+ encoding) for quality trimming. Default: ON.\r\n",
      "\r\n",
      "--phred64               Instructs Cutadapt to use ASCII+64 quality scores as Phred scores\r\n",
      "                        (Illumina 1.5 encoding) for quality trimming.\r\n",
      "\r\n",
      "--fastqc                Run FastQC in the default mode on the FastQ file once trimming is complete.\r\n",
      "\r\n",
      "--fastqc_args \"<ARGS>\"  Passes extra arguments to FastQC. If more than one argument is to be passed\r\n",
      "                        to FastQC they must be in the form \"arg1 arg2 etc.\". An example would be:\r\n",
      "                        --fastqc_args \"--nogroup --outdir /home/\". Passing extra arguments will\r\n",
      "                        automatically invoke FastQC, so --fastqc does not have to be specified\r\n",
      "                        separately.\r\n",
      "\r\n",
      "-a/--adapter <STRING>   Adapter sequence to be trimmed. If not specified explicitly, Trim Galore will\r\n",
      "                        try to auto-detect whether the Illumina universal, Nextera transposase or Illumina\r\n",
      "                        small RNA adapter sequence was used. Also see '--illumina', '--nextera' and\r\n",
      "                        '--small_rna'. If no adapter can be detected within the first 1 million sequences\r\n",
      "                        of the first file specified Trim Galore defaults to '--illumina'.\r\n",
      "\r\n",
      "-a2/--adapter2 <STRING> Optional adapter sequence to be trimmed off read 2 of paired-end files. This\r\n",
      "                        option requires '--paired' to be specified as well. If the libraries to be trimmed\r\n",
      "                        are smallRNA then a2 will be set to the Illumina small RNA 5' adapter automatically\r\n",
      "                        (GATCGTCGGACT).\r\n",
      "\r\n",
      "--illumina              Adapter sequence to be trimmed is the first 13bp of the Illumina universal adapter\r\n",
      "                        'AGATCGGAAGAGC' instead of the default auto-detection of adapter sequence.\r\n",
      "\r\n",
      "--nextera               Adapter sequence to be trimmed is the first 12bp of the Nextera adapter\r\n",
      "                        'CTGTCTCTTATA' instead of the default auto-detection of adapter sequence.\r\n",
      "\r\n",
      "--small_rna             Adapter sequence to be trimmed is the first 12bp of the Illumina Small RNA 3' Adapter\r\n",
      "                        'TGGAATTCTCGG' instead of the default auto-detection of adapter sequence. Selecting\r\n",
      "                        to trim smallRNA adapters will also lower the --length value to 18bp. If the smallRNA\r\n",
      "                        libraries are paired-end then a2 will be set to the Illumina small RNA 5' adapter\r\n",
      "                        automatically (GATCGTCGGACT) unless -a 2 had been defined explicitly.\r\n",
      "\r\n",
      "--max_length <INT>      Discard reads that are longer than <INT> bp after trimming. This is only advised for\r\n",
      "                        smallRNA sequencing to remove non-small RNA sequences.\r\n",
      "\r\n",
      "\r\n",
      "--stringency <INT>      Overlap with adapter sequence required to trim a sequence. Defaults to a\r\n",
      "                        very stringent setting of 1, i.e. even a single bp of overlapping sequence\r\n",
      "                        will be trimmed off from the 3' end of any read.\r\n",
      "\r\n",
      "-e <ERROR RATE>         Maximum allowed error rate (no. of errors divided by the length of the matching\r\n",
      "                        region) (default: 0.1)\r\n",
      "\r\n",
      "--gzip                  Compress the output file with GZIP. If the input files are GZIP-compressed\r\n",
      "                        the output files will automatically be GZIP compressed as well. As of v0.2.8 the\r\n",
      "                        compression will take place on the fly.\r\n",
      "\r\n",
      "--dont_gzip             Output files won't be compressed with GZIP. This option overrides --gzip.\r\n",
      "\r\n",
      "--length <INT>          Discard reads that became shorter than length INT because of either\r\n",
      "                        quality or adapter trimming. A value of '0' effectively disables\r\n",
      "                        this behaviour. Default: 20 bp.\r\n",
      "\r\n",
      "                        For paired-end files, both reads of a read-pair need to be longer than\r\n",
      "                        <INT> bp to be printed out to validated paired-end files (see option --paired).\r\n",
      "                        If only one read became too short there is the possibility of keeping such\r\n",
      "                        unpaired single-end reads (see --retain_unpaired). Default pair-cutoff: 20 bp.\r\n",
      "\r\n",
      "--max_n COUNT           The total number of Ns (as integer) a read may contain before it will be removed altogether.\r\n",
      "                        In a paired-end setting, either read exceeding this limit will result in the entire\r\n",
      "                        pair being removed from the trimmed output files.\r\n",
      "\r\n",
      "--trim-n                Removes Ns from either side of the read. This option does currently not work in RRBS mode.\r\n",
      "\r\n",
      "-o/--output_dir <DIR>   If specified all output will be written to this directory instead of the current\r\n",
      "                        directory.\r\n",
      "\r\n",
      "--no_report_file        If specified no report file will be generated.\r\n",
      "\r\n",
      "--suppress_warn         If specified any output to STDOUT or STDERR will be suppressed.\r\n",
      "\r\n",
      "--clip_R1 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 1 (or single-end\r\n",
      "                        reads). This may be useful if the qualities were very poor, or if there is some\r\n",
      "                        sort of unwanted bias at the 5' end. Default: OFF.\r\n",
      "\r\n",
      "--clip_R2 <int>         Instructs Trim Galore to remove <int> bp from the 5' end of read 2 (paired-end reads\r\n",
      "                        only). This may be useful if the qualities were very poor, or if there is some sort\r\n",
      "                        of unwanted bias at the 5' end. For paired-end BS-Seq, it is recommended to remove\r\n",
      "                        the first few bp because the end-repair reaction may introduce a bias towards low\r\n",
      "                        methylation. Please refer to the M-bias plot section in the Bismark User Guide for\r\n",
      "                        some examples. Default: OFF.\r\n",
      "\r\n",
      "--three_prime_clip_R1 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 1 (or single-end\r\n",
      "                        reads) AFTER adapter/quality trimming has been performed. This may remove some unwanted\r\n",
      "                        bias from the 3' end that is not directly related to adapter sequence or basecall quality.\r\n",
      "                        Default: OFF.\r\n",
      "\r\n",
      "--three_prime_clip_R2 <int>     Instructs Trim Galore to remove <int> bp from the 3' end of read 2 AFTER\r\n",
      "                        adapter/quality trimming has been performed. This may remove some unwanted bias from\r\n",
      "                        the 3' end that is not directly related to adapter sequence or basecall quality.\r\n",
      "                        Default: OFF.\r\n",
      "\r\n",
      "--path_to_cutadapt </path/to/cutadapt>     You may use this option to specify a path to the Cutadapt executable,\r\n",
      "                        e.g. /my/home/cutadapt-1.7.1/bin/cutadapt. Else it is assumed that Cutadapt is in\r\n",
      "                        the PATH.\r\n",
      "\r\n",
      "\r\n",
      "RRBS-specific options (MspI digested material):\r\n",
      "\r\n",
      "--rrbs                  Specifies that the input file was an MspI digested RRBS sample (recognition\r\n",
      "                        site: CCGG). Single-end or Read 1 sequences (paired-end) which were adapter-trimmed\r\n",
      "                        will have a further 2 bp removed from their 3' end. Sequences which were merely\r\n",
      "                        trimmed because of poor quality will not be shortened further. Read 2 of paired-end\r\n",
      "                        libraries will in addition have the first 2 bp removed from the 5' end (by setting \r\n",
      "                        '--clip_r2 2'). This is to avoid using artificial methylation calls from the filled-in\r\n",
      "                        cytosine positions close to the 3' MspI site in sequenced fragments. \r\n",
      "                        This option is not recommended for users of the NuGEN ovation RRBS System 1-16\r\n",
      "                        kit (see below).\r\n",
      "\r\n",
      "--non_directional       Selecting this option for non-directional RRBS libraries will screen\r\n",
      "                        quality-trimmed sequences for 'CAA' or 'CGA' at the start of the read\r\n",
      "                        and, if found, removes the first two basepairs. Like with the option\r\n",
      "                        '--rrbs' this avoids using cytosine positions that were filled-in\r\n",
      "                        during the end-repair step. '--non_directional' requires '--rrbs' to\r\n",
      "                        be specified as well. Note that this option does not set '--clip_r2 2' in\r\n",
      "                        paired-end mode.\r\n",
      "\r\n",
      "--keep                  Keep the quality trimmed intermediate file. Default: off, which means\r\n",
      "                        the temporary file is being deleted after adapter trimming. Only has\r\n",
      "                        an effect for RRBS samples since other FastQ files are not trimmed\r\n",
      "                        for poor qualities separately.\r\n",
      "\r\n",
      "\r\n",
      "Note for RRBS using the NuGEN Ovation RRBS System 1-16 kit:\r\n",
      "\r\n",
      "Owing to the fact that the NuGEN Ovation kit attaches a varying number of nucleotides (0-3) after each MspI\r\n",
      "site Trim Galore should be run WITHOUT the option --rrbs. This trimming is accomplished in a subsequent \r\n",
      "diversity trimming step afterwards (see their manual).\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Note for RRBS using MseI:\r\n",
      "\r\n",
      "If your DNA material was digested with MseI (recognition motif: TTAA) instead of MspI it is NOT necessary\r\n",
      "to specify --rrbs or --non_directional since virtually all reads should start with the sequence\r\n",
      "'TAA', and this holds true for both directional and non-directional libraries. As the end-repair of 'TAA'\r\n",
      "restricted sites does not involve any cytosines it does not need to be treated especially. Instead, simply\r\n",
      "run Trim Galore! in the standard (i.e. non-RRBS) mode.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "Paired-end specific options:\r\n",
      "\r\n",
      "--paired                This option performs length trimming of quality/adapter/RRBS trimmed reads for\r\n",
      "                        paired-end files. To pass the validation test, both sequences of a sequence pair\r\n",
      "                        are required to have a certain minimum length which is governed by the option\r\n",
      "                        --length (see above). If only one read passes this length threshold the\r\n",
      "                        other read can be rescued (see option --retain_unpaired). Using this option lets\r\n",
      "                        you discard too short read pairs without disturbing the sequence-by-sequence order\r\n",
      "                        of FastQ files which is required by many aligners.\r\n",
      "\r\n",
      "                        Trim Galore! expects paired-end files to be supplied in a pairwise fashion, e.g.\r\n",
      "                        file1_1.fq file1_2.fq SRR2_1.fq.gz SRR2_2.fq.gz ... .\r\n",
      "\r\n",
      "-t/--trim1              Trims 1 bp off every read from its 3' end. This may be needed for FastQ files that\r\n",
      "                        are to be aligned as paired-end data with Bowtie. This is because Bowtie (1) regards\r\n",
      "                        alignments like this:\r\n",
      "\r\n",
      "                          R1 --------------------------->     or this:    ----------------------->  R1\r\n",
      "                          R2 <---------------------------                       <-----------------  R2\r\n",
      "\r\n",
      "                        as invalid (whenever a start/end coordinate is contained within the other read).\r\n",
      "                        NOTE: If you are planning to use Bowtie2, BWA etc. you don't need to specify this option.\r\n",
      "\r\n",
      "--retain_unpaired       If only one of the two paired-end reads became too short, the longer\r\n",
      "                        read will be written to either '.unpaired_1.fq' or '.unpaired_2.fq'\r\n",
      "                        output files. The length cutoff for unpaired single-end reads is\r\n",
      "                        governed by the parameters -r1/--length_1 and -r2/--length_2. Default: OFF.\r\n",
      "\r\n",
      "-r1/--length_1 <INT>    Unpaired single-end read length cutoff needed for read 1 to be written to\r\n",
      "                        '.unpaired_1.fq' output file. These reads may be mapped in single-end mode.\r\n",
      "                        Default: 35 bp.\r\n",
      "\r\n",
      "-r2/--length_2 <INT>    Unpaired single-end read length cutoff needed for read 2 to be written to\r\n",
      "                        '.unpaired_2.fq' output file. These reads may be mapped in single-end mode.\r\n",
      "                        Default: 35 bp.\r\n",
      "\r\n",
      "\r\n",
      "Last modified on 13 November 2017.\r\n",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/TrimGalore-0.4.5/trim_galore -help\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-09-21T23:19:08.597Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No quality encoding type selected. Assuming that the data provided uses Sanger encoded Phred scores (default)\n",
      "\n",
      "Path to Cutadapt set as: 'cutadapt' (default)\n",
      "1.16\n",
      "Cutadapt seems to be working fine (tested command 'cutadapt --version')\n",
      "\n",
      "\n",
      "AUTO-DETECTING ADAPTER TYPE\n",
      "===========================\n",
      "Attempting to auto-detect adapter type from the first 1 million sequences of the first file (>> /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz <<)\n",
      "\n",
      "Found perfect matches for the following adapter sequences:\n",
      "Adapter type\tCount\tSequence\tSequences analysed\tPercentage\n",
      "Illumina\t0\tAGATCGGAAGAGC\t1000000\t0.00\n",
      "smallRNA\t0\tTGGAATTCTCGG\t1000000\t0.00\n",
      "Nextera\t0\tCTGTCTCTTATA\t1000000\t0.00\n",
      "Unable to auto-detect most prominent adapter from the first specified file (count Illumina: 0, count smallRNA: 0)\n",
      "Defaulting to Illumina universal adapter ( AGATCGGAAGAGC ). Specify -a SEQUENCE to avoid this behavior).\n",
      "\n",
      "gunzip: error writing to output: Broken pipe\n",
      "gunzip: /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz: uncompress failed\n",
      "Writing report to '/Volumes/block/caligus/trim/Sealice_F1_S20_L001_R1_001_val_1.fq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.4_dev\n",
      "Cutadapt version: 1.16\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "All Read 1 sequences will be trimmed by 7 bp from their 5' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 7 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n",
      "All Read 1 sequences will be trimmed by 7 bp from their 3' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 7 bp from their 3' end to avoid poor qualities or biases\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to Sealice_F1_S20_L001_R1_001_val_1_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz <<< \n",
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "70000000 sequences processed\n",
      "80000000 sequences processed\n",
      "90000000 sequences processed\n",
      "100000000 sequences processed\n",
      "110000000 sequences processed\n",
      "This is cutadapt 1.16 with Python 3.6.8\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 14125.09 s (125 us/read; 0.48 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:             113,253,920\n",
      "Reads with adapters:                32,901,059 (29.1%)\n",
      "Reads written (passing filters):   113,253,920 (100.0%)\n",
      "\n",
      "Total basepairs processed: 13,839,041,941 bp\n",
      "Quality-trimmed:               4,233,820 bp (0.0%)\n",
      "Total written (filtered):  13,791,013,279 bp (99.7%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 32901059 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 46.9%\n",
      "  C: 10.9%\n",
      "  G: 14.0%\n",
      "  T: 28.2%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t27249332\t28313480.0\t0\t27249332\n",
      "2\t3756099\t7078370.0\t0\t3756099\n",
      "3\t1353389\t1769592.5\t0\t1353389\n",
      "4\t339506\t442398.1\t0\t339506\n",
      "5\t10944\t110599.5\t0\t10944\n",
      "6\t3619\t27649.9\t0\t3619\n",
      "7\t756\t6912.5\t0\t756\n",
      "8\t2137\t1728.1\t0\t2137\n",
      "9\t1436\t432.0\t0\t600 836\n",
      "10\t5243\t108.0\t1\t3042 2201\n",
      "11\t3617\t27.0\t1\t690 2927\n",
      "12\t6645\t6.8\t1\t1430 5215\n",
      "13\t9003\t1.7\t1\t0 9003\n",
      "14\t11117\t1.7\t1\t0 11117\n",
      "15\t22931\t1.7\t1\t0 22931\n",
      "16\t57394\t1.7\t1\t0 57394\n",
      "17\t43251\t1.7\t1\t0 43251\n",
      "18\t3691\t1.7\t1\t0 3691\n",
      "19\t640\t1.7\t1\t0 640\n",
      "20\t422\t1.7\t1\t0 422\n",
      "21\t457\t1.7\t1\t0 457\n",
      "22\t442\t1.7\t1\t0 442\n",
      "23\t456\t1.7\t1\t0 456\n",
      "24\t517\t1.7\t1\t0 517\n",
      "25\t639\t1.7\t1\t0 639\n",
      "26\t712\t1.7\t1\t0 712\n",
      "27\t481\t1.7\t1\t0 481\n",
      "28\t504\t1.7\t1\t0 504\n",
      "29\t474\t1.7\t1\t0 474\n",
      "30\t541\t1.7\t1\t0 541\n",
      "31\t595\t1.7\t1\t0 595\n",
      "32\t522\t1.7\t1\t0 522\n",
      "33\t468\t1.7\t1\t0 468\n",
      "34\t409\t1.7\t1\t0 409\n",
      "35\t312\t1.7\t1\t0 312\n",
      "36\t282\t1.7\t1\t0 282\n",
      "37\t289\t1.7\t1\t0 289\n",
      "38\t259\t1.7\t1\t0 259\n",
      "39\t280\t1.7\t1\t0 280\n",
      "40\t258\t1.7\t1\t0 258\n",
      "41\t286\t1.7\t1\t0 286\n",
      "42\t272\t1.7\t1\t0 272\n",
      "43\t247\t1.7\t1\t0 247\n",
      "44\t287\t1.7\t1\t0 287\n",
      "45\t268\t1.7\t1\t0 268\n",
      "46\t299\t1.7\t1\t0 299\n",
      "47\t264\t1.7\t1\t0 264\n",
      "48\t256\t1.7\t1\t0 256\n",
      "49\t304\t1.7\t1\t0 304\n",
      "50\t251\t1.7\t1\t0 251\n",
      "51\t278\t1.7\t1\t0 278\n",
      "52\t224\t1.7\t1\t0 224\n",
      "53\t238\t1.7\t1\t0 238\n",
      "54\t252\t1.7\t1\t0 252\n",
      "55\t250\t1.7\t1\t0 250\n",
      "56\t277\t1.7\t1\t0 277\n",
      "57\t224\t1.7\t1\t0 224\n",
      "58\t236\t1.7\t1\t0 236\n",
      "59\t231\t1.7\t1\t0 231\n",
      "60\t242\t1.7\t1\t0 242\n",
      "61\t229\t1.7\t1\t0 229\n",
      "62\t240\t1.7\t1\t0 240\n",
      "63\t212\t1.7\t1\t0 212\n",
      "64\t224\t1.7\t1\t0 224\n",
      "65\t189\t1.7\t1\t0 189\n",
      "66\t185\t1.7\t1\t0 185\n",
      "67\t230\t1.7\t1\t0 230\n",
      "68\t214\t1.7\t1\t0 214\n",
      "69\t198\t1.7\t1\t0 198\n",
      "70\t201\t1.7\t1\t0 201\n",
      "71\t186\t1.7\t1\t0 186\n",
      "72\t191\t1.7\t1\t0 191\n",
      "73\t162\t1.7\t1\t0 162\n",
      "74\t165\t1.7\t1\t0 165\n",
      "75\t169\t1.7\t1\t0 169\n",
      "76\t168\t1.7\t1\t0 168\n",
      "77\t151\t1.7\t1\t0 151\n",
      "78\t154\t1.7\t1\t0 154\n",
      "79\t117\t1.7\t1\t0 117\n",
      "80\t141\t1.7\t1\t0 141\n",
      "81\t106\t1.7\t1\t0 106\n",
      "82\t109\t1.7\t1\t0 109\n",
      "83\t108\t1.7\t1\t0 108\n",
      "84\t102\t1.7\t1\t0 102\n",
      "85\t114\t1.7\t1\t0 114\n",
      "86\t102\t1.7\t1\t0 102\n",
      "87\t109\t1.7\t1\t0 109\n",
      "88\t105\t1.7\t1\t0 105\n",
      "89\t105\t1.7\t1\t0 105\n",
      "90\t81\t1.7\t1\t0 81\n",
      "91\t82\t1.7\t1\t0 82\n",
      "92\t82\t1.7\t1\t0 82\n",
      "93\t52\t1.7\t1\t0 52\n",
      "94\t63\t1.7\t1\t0 63\n",
      "95\t46\t1.7\t1\t0 46\n",
      "96\t53\t1.7\t1\t0 53\n",
      "97\t56\t1.7\t1\t0 56\n",
      "98\t56\t1.7\t1\t0 56\n",
      "99\t48\t1.7\t1\t0 48\n",
      "100\t44\t1.7\t1\t0 44\n",
      "101\t53\t1.7\t1\t0 53\n",
      "102\t38\t1.7\t1\t0 38\n",
      "103\t48\t1.7\t1\t0 48\n",
      "104\t27\t1.7\t1\t0 27\n",
      "105\t38\t1.7\t1\t0 38\n",
      "106\t29\t1.7\t1\t0 29\n",
      "107\t29\t1.7\t1\t0 29\n",
      "108\t22\t1.7\t1\t0 22\n",
      "109\t18\t1.7\t1\t0 18\n",
      "110\t17\t1.7\t1\t0 17\n",
      "111\t8\t1.7\t1\t0 8\n",
      "112\t12\t1.7\t1\t0 12\n",
      "113\t11\t1.7\t1\t0 11\n",
      "114\t15\t1.7\t1\t0 15\n",
      "115\t8\t1.7\t1\t0 8\n",
      "116\t18\t1.7\t1\t0 18\n",
      "117\t9\t1.7\t1\t0 9\n",
      "118\t14\t1.7\t1\t0 14\n",
      "119\t11\t1.7\t1\t0 11\n",
      "120\t5\t1.7\t1\t0 5\n",
      "121\t4\t1.7\t1\t0 4\n",
      "122\t3\t1.7\t1\t0 3\n",
      "123\t4\t1.7\t1\t0 4\n",
      "124\t4\t1.7\t1\t0 4\n",
      "125\t2\t1.7\t1\t0 2\n",
      "126\t2\t1.7\t1\t0 2\n",
      "127\t3\t1.7\t1\t0 3\n",
      "128\t1\t1.7\t1\t0 1\n",
      "130\t1\t1.7\t1\t0 1\n",
      "131\t1\t1.7\t1\t0 1\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz\n",
      "=============================================\n",
      "113253920 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Writing report to '/Volumes/block/caligus/trim/Sealice_F1_S20_L001_R2_001_val_2.fq.gz_trimming_report.txt'\n",
      "\n",
      "SUMMARISING RUN PARAMETERS\n",
      "==========================\n",
      "Input filename: /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R2_001_val_2.fq.gz\n",
      "Trimming mode: paired-end\n",
      "Trim Galore version: 0.4.4_dev\n",
      "Cutadapt version: 1.16\n",
      "Quality Phred score cutoff: 20\n",
      "Quality encoding type selected: ASCII+33\n",
      "Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; default (inconclusive auto-detection))\n",
      "Maximum trimming error rate: 0.1 (default)\n",
      "Minimum required adapter overlap (stringency): 1 bp\n",
      "Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp\n",
      "All Read 1 sequences will be trimmed by 7 bp from their 5' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 7 bp from their 5' end to avoid poor qualities or biases (e.g. M-bias for BS-Seq applications)\n",
      "All Read 1 sequences will be trimmed by 7 bp from their 3' end to avoid poor qualities or biases\n",
      "All Read 2 sequences will be trimmed by 7 bp from their 3' end to avoid poor qualities or biases\n",
      "Output file(s) will be GZIP compressed\n",
      "\n",
      "Writing final adapter and quality trimmed output to Sealice_F1_S20_L001_R2_001_val_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      "  >>> Now performing quality (cutoff 20) and adapter trimming in a single pass for the adapter sequence: 'AGATCGGAAGAGC' from file /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R2_001_val_2.fq.gz <<< \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10000000 sequences processed\n",
      "20000000 sequences processed\n",
      "30000000 sequences processed\n",
      "40000000 sequences processed\n",
      "50000000 sequences processed\n",
      "60000000 sequences processed\n",
      "70000000 sequences processed\n",
      "80000000 sequences processed\n",
      "90000000 sequences processed\n",
      "100000000 sequences processed\n",
      "110000000 sequences processed\n",
      "This is cutadapt 1.16 with Python 3.6.8\n",
      "Command line parameters: -f fastq -e 0.1 -q 20 -O 1 -a AGATCGGAAGAGC /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R2_001_val_2.fq.gz\n",
      "Running on 1 core\n",
      "Trimming 1 adapter with at most 10.0% errors in single-end mode ...\n",
      "Finished in 43396.28 s (383 us/read; 0.16 M reads/minute).\n",
      "\n",
      "=== Summary ===\n",
      "\n",
      "Total reads processed:             113,253,920\n",
      "Reads with adapters:                34,655,811 (30.6%)\n",
      "Reads written (passing filters):   113,253,920 (100.0%)\n",
      "\n",
      "Total basepairs processed: 13,863,655,209 bp\n",
      "Quality-trimmed:               9,198,051 bp (0.1%)\n",
      "Total written (filtered):  13,807,908,392 bp (99.6%)\n",
      "\n",
      "=== Adapter 1 ===\n",
      "\n",
      "Sequence: AGATCGGAAGAGC; Type: regular 3'; Length: 13; Trimmed: 34655811 times.\n",
      "\n",
      "No. of allowed errors:\n",
      "0-9 bp: 0; 10-13 bp: 1\n",
      "\n",
      "Bases preceding removed adapters:\n",
      "  A: 49.8%\n",
      "  C: 11.2%\n",
      "  G: 12.6%\n",
      "  T: 26.4%\n",
      "  none/other: 0.0%\n",
      "\n",
      "Overview of removed sequences\n",
      "length\tcount\texpect\tmax.err\terror counts\n",
      "1\t28202874\t28313480.0\t0\t28202874\n",
      "2\t4517996\t7078370.0\t0\t4517996\n",
      "3\t1382508\t1769592.5\t0\t1382508\n",
      "4\t316842\t442398.1\t0\t316842\n",
      "5\t10124\t110599.5\t0\t10124\n",
      "6\t3888\t27649.9\t0\t3888\n",
      "7\t950\t6912.5\t0\t950\n",
      "8\t3434\t1728.1\t0\t3434\n",
      "9\t2078\t432.0\t0\t859 1219\n",
      "10\t7426\t108.0\t1\t4140 3286\n",
      "11\t5381\t27.0\t1\t1087 4294\n",
      "12\t9914\t6.8\t1\t2122 7792\n",
      "13\t10263\t1.7\t1\t0 10263\n",
      "14\t15009\t1.7\t1\t0 15009\n",
      "15\t34126\t1.7\t1\t0 34126\n",
      "16\t72970\t1.7\t1\t0 72970\n",
      "17\t39934\t1.7\t1\t0 39934\n",
      "18\t3972\t1.7\t1\t0 3972\n",
      "19\t625\t1.7\t1\t0 625\n",
      "20\t374\t1.7\t1\t0 374\n",
      "21\t333\t1.7\t1\t0 333\n",
      "22\t300\t1.7\t1\t0 300\n",
      "23\t349\t1.7\t1\t0 349\n",
      "24\t471\t1.7\t1\t0 471\n",
      "25\t725\t1.7\t1\t0 725\n",
      "26\t587\t1.7\t1\t0 587\n",
      "27\t406\t1.7\t1\t0 406\n",
      "28\t441\t1.7\t1\t0 441\n",
      "29\t430\t1.7\t1\t0 430\n",
      "30\t455\t1.7\t1\t0 455\n",
      "31\t438\t1.7\t1\t0 438\n",
      "32\t458\t1.7\t1\t0 458\n",
      "33\t392\t1.7\t1\t0 392\n",
      "34\t328\t1.7\t1\t0 328\n",
      "35\t234\t1.7\t1\t0 234\n",
      "36\t217\t1.7\t1\t0 217\n",
      "37\t209\t1.7\t1\t0 209\n",
      "38\t201\t1.7\t1\t0 201\n",
      "39\t157\t1.7\t1\t0 157\n",
      "40\t179\t1.7\t1\t0 179\n",
      "41\t179\t1.7\t1\t0 179\n",
      "42\t181\t1.7\t1\t0 181\n",
      "43\t201\t1.7\t1\t0 201\n",
      "44\t171\t1.7\t1\t0 171\n",
      "45\t174\t1.7\t1\t0 174\n",
      "46\t182\t1.7\t1\t0 182\n",
      "47\t208\t1.7\t1\t0 208\n",
      "48\t190\t1.7\t1\t0 190\n",
      "49\t196\t1.7\t1\t0 196\n",
      "50\t194\t1.7\t1\t0 194\n",
      "51\t178\t1.7\t1\t0 178\n",
      "52\t193\t1.7\t1\t0 193\n",
      "53\t192\t1.7\t1\t0 192\n",
      "54\t162\t1.7\t1\t0 162\n",
      "55\t203\t1.7\t1\t0 203\n",
      "56\t173\t1.7\t1\t0 173\n",
      "57\t184\t1.7\t1\t0 184\n",
      "58\t165\t1.7\t1\t0 165\n",
      "59\t149\t1.7\t1\t0 149\n",
      "60\t162\t1.7\t1\t0 162\n",
      "61\t181\t1.7\t1\t0 181\n",
      "62\t172\t1.7\t1\t0 172\n",
      "63\t192\t1.7\t1\t0 192\n",
      "64\t166\t1.7\t1\t0 166\n",
      "65\t138\t1.7\t1\t0 138\n",
      "66\t117\t1.7\t1\t0 117\n",
      "67\t148\t1.7\t1\t0 148\n",
      "68\t151\t1.7\t1\t0 151\n",
      "69\t141\t1.7\t1\t0 141\n",
      "70\t139\t1.7\t1\t0 139\n",
      "71\t96\t1.7\t1\t0 96\n",
      "72\t138\t1.7\t1\t0 138\n",
      "73\t131\t1.7\t1\t0 131\n",
      "74\t133\t1.7\t1\t0 133\n",
      "75\t122\t1.7\t1\t0 122\n",
      "76\t107\t1.7\t1\t0 107\n",
      "77\t104\t1.7\t1\t0 104\n",
      "78\t91\t1.7\t1\t0 91\n",
      "79\t86\t1.7\t1\t0 86\n",
      "80\t92\t1.7\t1\t0 92\n",
      "81\t86\t1.7\t1\t0 86\n",
      "82\t85\t1.7\t1\t0 85\n",
      "83\t84\t1.7\t1\t0 84\n",
      "84\t88\t1.7\t1\t0 88\n",
      "85\t71\t1.7\t1\t0 71\n",
      "86\t64\t1.7\t1\t0 64\n",
      "87\t67\t1.7\t1\t0 67\n",
      "88\t54\t1.7\t1\t0 54\n",
      "89\t59\t1.7\t1\t0 59\n",
      "90\t61\t1.7\t1\t0 61\n",
      "91\t55\t1.7\t1\t0 55\n",
      "92\t45\t1.7\t1\t0 45\n",
      "93\t56\t1.7\t1\t0 56\n",
      "94\t36\t1.7\t1\t0 36\n",
      "95\t48\t1.7\t1\t0 48\n",
      "96\t43\t1.7\t1\t0 43\n",
      "97\t36\t1.7\t1\t0 36\n",
      "98\t35\t1.7\t1\t0 35\n",
      "99\t38\t1.7\t1\t0 38\n",
      "100\t38\t1.7\t1\t0 38\n",
      "101\t21\t1.7\t1\t0 21\n",
      "102\t27\t1.7\t1\t0 27\n",
      "103\t29\t1.7\t1\t0 29\n",
      "104\t37\t1.7\t1\t0 37\n",
      "105\t24\t1.7\t1\t0 24\n",
      "106\t18\t1.7\t1\t0 18\n",
      "107\t25\t1.7\t1\t0 25\n",
      "108\t12\t1.7\t1\t0 12\n",
      "109\t17\t1.7\t1\t0 17\n",
      "110\t26\t1.7\t1\t0 26\n",
      "111\t15\t1.7\t1\t0 15\n",
      "112\t12\t1.7\t1\t0 12\n",
      "113\t17\t1.7\t1\t0 17\n",
      "114\t13\t1.7\t1\t0 13\n",
      "115\t12\t1.7\t1\t0 12\n",
      "116\t7\t1.7\t1\t0 7\n",
      "117\t7\t1.7\t1\t0 7\n",
      "118\t10\t1.7\t1\t0 10\n",
      "119\t6\t1.7\t1\t0 6\n",
      "120\t9\t1.7\t1\t0 9\n",
      "121\t10\t1.7\t1\t0 10\n",
      "122\t8\t1.7\t1\t0 8\n",
      "123\t1\t1.7\t1\t0 1\n",
      "124\t8\t1.7\t1\t0 8\n",
      "125\t2\t1.7\t1\t0 2\n",
      "126\t6\t1.7\t1\t0 6\n",
      "129\t1\t1.7\t1\t0 1\n",
      "130\t2\t1.7\t1\t0 2\n",
      "\n",
      "\n",
      "RUN STATISTICS FOR INPUT FILE: /Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R2_001_val_2.fq.gz\n",
      "=============================================\n",
      "113253920 sequences processed in total\n",
      "The length threshold of paired-end sequences gets evaluated later on (in the validation step)\n",
      "\n",
      "Validate paired-end files Sealice_F1_S20_L001_R1_001_val_1_trimmed.fq.gz and Sealice_F1_S20_L001_R2_001_val_2_trimmed.fq.gz\n",
      "file_1: Sealice_F1_S20_L001_R1_001_val_1_trimmed.fq.gz, file_2: Sealice_F1_S20_L001_R2_001_val_2_trimmed.fq.gz\n",
      "\n",
      "\n",
      ">>>>> Now validing the length of the 2 paired-end infiles: Sealice_F1_S20_L001_R1_001_val_1_trimmed.fq.gz and Sealice_F1_S20_L001_R2_001_val_2_trimmed.fq.gz <<<<<\n",
      "Writing validated paired-end read 1 reads to Sealice_F1_S20_L001_R1_001_val_1_val_1.fq.gz\n",
      "Writing validated paired-end read 2 reads to Sealice_F1_S20_L001_R2_001_val_2_val_2.fq.gz\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/TrimGalore-0.4.5/trim_galore \\\n",
    "--paired \\\n",
    "--clip_R1 7 \\\n",
    "--clip_R2 7 \\\n",
    "--three_prime_clip_R1 7 \\\n",
    "--three_prime_clip_R2 7 \\\n",
    "--output_dir /Volumes/block/caligus/trim/ \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L001_R2_001_val_2.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L002_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F1_S20_L002_R2_001_val_2.fq.gz \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2019-09-21T23:20:49.366Z"
    }
   },
   "outputs": [],
   "source": [
    "!/Applications/bioinfo/TrimGalore-0.4.5/trim_galore \\\n",
    "--paired \\\n",
    "--clip_R1 7 \\\n",
    "--clip_R2 7 \\\n",
    "--three_prime_clip_R1 7 \\\n",
    "--three_prime_clip_R2 7 \\\n",
    "--output_dir /Volumes/block/caligus/trim/ \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L001_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L001_R2_001_val_2.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L002_R1_001_val_1.fq.gz \\\n",
    "/Volumes/web/metacarcinus/Salmo_Calig/analyses/20190806_TrimGalore/Sealice_F2_S22_L002_R2_001_val_2.fq.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
